pkg/publish/ingress.go

```diff
@@ -21,27 +24,58 @@ import (
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func toPrefix(domain, serviceName string, appInstance *v1.AppInstance) (hostPrefix string) {
-	if strings.HasSuffix(domain, "on-acorn.io") {
-		var appInstanceIDSegment string
-		var appInstanceIDSegmentByte [32]byte
+var ErrPatternParseFailed = errors.New("failed to parse endpoint pattern")
 
-		appInstanceIDSegment = serviceName + ":" + appInstance.GetName() + ":" + appInstance.Namespace
-		appInstanceIDSegmentByte = sha256.Sum256([]byte(appInstanceIDSegment))
-		appInstanceIDSegment = hex.EncodeToString(appInstanceIDSegmentByte[:])[:12]
-		hostPrefix = name.Limit(serviceName+"-"+appInstance.GetName(), 62-len(appInstanceIDSegment)) + "-" + appInstanceIDSegment
+func toEndpoint(domain, serviceName string, appInstance *v1.AppInstance) (endpoint string, err error) {
+	domain = strings.TrimPrefix(domain, ".")
+	pattern := appInstance.Spec.HttpEndpointPattern
+	if pattern != "" {
+		test := struct {
```

Comment:
nit: test feels like a weird name?

Reply:
Left over from debugging stuff, got it all fixed up now.